api/rooms.js

```diff
@@ -0,0 +1,82 @@
+const express = require("express");
+const { ObjectId } = require("mongodb");
+
+module.exports = function rooms(db) {
+  const router = express.Router();
+
+  const usersCollection = db.collection("users");
+  const roomsCollection = db.collection("rooms");
+
+  // GET
+  router.get("/", function (req, res) {
+    roomsCollection.find().toArray((err, userDocs) => {
+        res.send({ data: userDocs.map(encodeRoom) });
+      });
+  });
+
+  // POST
+  router.post("/", function (req, res, next) {
+    let userIds = req.body.participants;
+
+    try { userIds = userIds.map(userId => new ObjectId(userId)); } catch(e) {console.log('Try Catch: ' + e)}
+
+    usersCollection.findOne({ _id: userIds[0]}, (err, userDoc) => {
+      if (err) {
+        next(err);
+      } else if (!userDoc) {
+        res.sendStatus(404);
+        return;
+      }
+    });
+
+    usersCollection.findOne({ _id: userIds[1]}, (err, userDoc) => {
+      if (err) {
+        next(err);
+      } else if (!userDoc) {
+        res.sendStatus(404);
+        return;
+      } 
+    });
+    
+    roomsCollection.insertOne({ createdAt: new Date(), participants: userIds }, (err, results) => {
+        if (err) {
+          next(err);
+        } 
+        res.json({ data: encodeRoom(results.ops[0]) }).status(201);
+      });
```

Comment:
There are a few problems here:

1. It's inefficient to make multiple calls to the database when one would suffice
2. You're only checking the first two user IDs, but there's nothing preventing more than one user ID being added to a room.
3. Because calls to `findOne` return right away, the call to `insertOne` will be called regarless of whether or not the users are found.

What I would suggest to solve these issues are:

1. Merge the user queries into one call to `usersCollection.find`
2. Use `_id: { $in: userIds}` instead of `_id: userIds[i]`
3. Call `roomsCollection.insertOne` in the callback for `usersCollection.find`

```js
usersCollection.find({ _id: { $in: userIds } }, (err, userDocs) => {
  if (err) {
    next(err);
  } else if (userDocs.length !== userIds.length) {
    res.json({ message: "Invalid user IDs" }).status(400); // use 400, not 404
  } else {
    roomsCollection.insertOne({ createdAt: new Date(), participants: userIds }, (err, results) => {
      if (err) {
        next(err);
      } else {
        res.json({ data: encodeRoom(results.ops[0]) }).status(201);
      }
    });
  }
});
```